feat(gnokey): -broadcast=true by default

jefft0 opened the pull request:
This PR replaces https://github.com/gnolang/gno/pull/3725 after discussion.

* In the gnokey `maketx.go`, change the `-broadcast` flag to default true
* In gnoweb, update `command.html` so that the help page gnokey commands are correct
* Update examples in the markdown files
* Update txtar files to remove `-broadcast` or add `-broadcast=false` where needed. (This is most of the changed files.)
* I didn't remove the `-broadcast` flag from [this script](https://github.com/gnolang/gno/blob/627790bee8346069775d3c1dc6b7861b6d84d0a4/misc/e2e/run_tests.sh#L70) because I'm not sure that it will run on a machine with the updated `gnokey`. (It doesn't hurt to leave the flag.)
* I didn't change the folders `gno.land/genesis` or `misc/deployments`

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🟢 Changes related to gnoweb must be reviewed by its codeowners
 🟢 Pending initial approval by a review team member, or review from tech-staff


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: jefft0/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Changes related to gnoweb must be reviewed by its codeowners</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 A changed file matches this pattern: ^gno.land/pkg/gnoweb/ (filename: gno.land/pkg/gnoweb/components/ui/command.html)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Or
    ├── 🔴 Or
    │   ├── 🔴 And
    │   │   ├── 🔴 Pull request author is user: alexiscolin
    │   │   └── 🟢 This user reviewed pull request: gfanton (with state "APPROVED")
    │   └── 🔴 And
    │       ├── 🔴 Pull request author is user: gfanton
    │       └── 🟢 This user reviewed pull request: alexiscolin (with state "APPROVED")
    └── 🟢 And
        ├── 🟢 Not (🔴 Pull request author is user: alexiscolin)
        ├── 🟢 Not (🔴 Pull request author is user: gfanton)
        └── 🟢 Or
            ├── 🟢 This user reviewed pull request: alexiscolin (with state "APPROVED")
            └── 🟢 This user reviewed pull request: gfanton (with state "APPROVED")

```
</blockquote></details>

<details><summary>Pending initial approval by a review team member, or review from tech-staff</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Not (🔴 Pull request author is a member of the team: tech-staff)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 If
    ├── 🟢 Condition
    │   └── 🟢 Or
    │       ├── 🟢 User notJoon already reviewed PR 4965 with state APPROVED
    │       ├── 🟢 At least 1 user(s) of the team tech-staff reviewed pull request
    │       └── 🔴 This pull request is a draft
    └── 🟢 Then
        └── 🟢 Not (🔴 This label is applied to pull request: review/triage-pending)

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

notJoon reviewed (approved):
`broadcast` has never been false for me either. I checked that the defauly value was changed in the CLI options properly.

aeddi reviewed (commented):
Looks weird to have a flag of this type set to `true` by default and to have to pass `--broadcast=false` to use it.
It would feel more idiomatic to flip this flag to a `--no-broadcast` or another more appropriate term.

davd-gzl reviewed (commented):
I agree `--no-broadcast` would be more idiomatic, but notice it might break older scripts.

---

You forgot to update a documentation:

### `docs/users/interact-with-gnokey.md`:
- **L149**: Add `(default: true)` to `-broadcast` description
- **L572**: Change "we didn't use the `-broadcast` flag" → "we used `-broadcast=false`"

And to remove the following broadcast from this txtars for consistency:
- gno.land/pkg/integration/testdata/interrealm_final_omarsy.txtar
- gno.land/pkg/integration/testdata/maketx_run_send.txtar
- gno.land/pkg/integration/testdata/banker_security.txtar
- gno.land/pkg/integration/testdata/restart_gas.txtar
- gno.land/pkg/integration/testdata/interrealm_final.txtar
- gno.land/pkg/integration/testdata/maketx_call_send.txtar
- gno.land/pkg/integration/testdata/panic.txtar
- gno.land/pkg/integration/testdata/issue_1543.txtar
- gno.land/pkg/integration/testdata/issue_4280.txtar
- gno.land/pkg/integration/testdata/issue_4280a.txtar
- gno.land/pkg/integration/testdata/issue_2266.txtar
- gno.land/pkg/integration/testdata/mapkey.txtar
- gno.land/pkg/integration/testdata/issue_4280b.txtar

mvallenet reviewed (approved):
Don't have strong opinion on broadcast default true or no-broadcast, the implem lgtm

jefft0 committed:
feat(gnokey): -broadcast=true by default

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

jefft0 committed:
Merge remote-tracking branch 'upstream/master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Rerun tests

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
chore: Remote -broadcast in txtar files

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

jefft0 committed:
Rerun tests

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Rerun tests

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

gfanton reviewed (approved):
I agree `-no-broadcast` is better, but this PR has been waiting long enough and the end result is the same default. Let's merge this and add `-no-broadcast` as a direct follow-up, unless @jefft0 you want to implement it right away.

Invoking @thehowl for a last call, but I think we should merge it.

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
chore: Remote -broadcast in txtar files

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

jefft0 committed:
re-run CI checks

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
chore: Remove -broadcast in txtar files

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

thehowl committed:
merge: upstream master into feat/gnokey-broadcast-default-true

Resolve conflict in maketx_call_send.txtar: upstream replaced the file
with a new realm.SentCoins() test; apply the PR's -broadcast removal to it.

thehowl reviewed (approved):
I've had a green-light by @moul to merge this as it improves UX drastically. Merging after CI is green.

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true

jefft0 committed:
chore: Remove -broadcast in txtar files

Signed-off-by: Jeff Thompson <jeff@thefirst.org>

jefft0 committed:
Merge branch 'master' into feat/gnokey-broadcast-default-true
